addons/website/models/website_visitor.py

```diff
@@ -28,14 +28,28 @@ class WebsiteTrack(models.Model):
 class WebsiteVisitor(models.Model):
     _name = 'website.visitor'
     _description = 'Website Visitor'
-    _order = 'last_connection_datetime DESC'
+    _order = 'id DESC'
 
-    name = fields.Char('Name')
-    access_token = fields.Char(required=True, default=lambda x: uuid.uuid4().hex, copy=False, groups='website.group_website_publisher')
-    active = fields.Boolean('Active', default=True)
+    def _get_access_token(self):
+        """ Either the user's partner.id or a hash. """
+        assert request, _("Visitors can only be created through the frontend.")
+
+        if not self.env.user._is_public():
+            return self.env.user.partner_id.id
+
+        msg = repr((
+            request.httprequest.remote_addr,
+            request.httprequest.environ.get('HTTP_USER_AGENT'),
+            # For testing purpose, uncomment this line if you want different
+            # URL (domain/ip) to generate different visitor.
+            # request.httprequest.host,
+        )).encode('utf-8')
+        return hashlib.sha1(msg).hexdigest()
```

Comment:
I assume this is only indicative for stats collection? Because this is pretty much under attacker control.

I'd suggest renaming to something other than "access token" as access token are usually considered to be... individual secret tokens which give access to resources. If this is just a visitor id / tag it's confusing and dangerous to call it an access token because it can easily lead to misuse both ways.